### DISCUSSION TITLE
E2 metrics

### DISCUSSION THREAD

**Discussion by apperezG** [srsran/srsRAN_Project#407] [General Help] [answered]
Good afternoon!

I'm configuring a deployment with srsRAN Project and FlexRIC. I'm trying to obtain different metrics to introduce them as input features to a neural network. 

In srsRAN Project there is a file that lists the available metrics [https://github.com/srsran/srsRAN_Project/blob/main/lib/e2/e2sm/e2sm_kpm/e2sm_kpm_du_meas_provider_impl.cpp](url). For now I am using the CQI, RSRP and RSRQ. 

But looking at this file it seems that both e2sm_kpm_du_meas_meas_provider_impl::get_rsrp (RSRP value) and e2sm_kpm_du_meas_provider_impl::get_rsrq (RSRQ value) get the same metric named ue_metrics.pusch_snr_db. Is this a bug or is it obtained this way for some reason? 

Another question I have is: Do you plan to introduce more metrics in the near future?

Thank you in advance

**Comment by pgawlowicz** [comment, marked as answer]
Hi,
those  CQI, RSRP, and RSRQ are some dummy metrics, that will be removed soon. We are currently working to add more metrics that are described in the ORAN documents - [here](https://github.com/srsran/srsRAN_Project/blob/main/lib/e2/e2sm/e2sm_kpm/e2sm_kpm_metric_defs.h) we will all metrics available in the ORAN documents. 

The metrics that are currently implemented are listed [here](https://github.com/srsran/srsRAN_Project/blob/main/lib/e2/e2sm/e2sm_kpm/e2sm_kpm_du_meas_provider_impl.cpp#L29).

More metrics is coming soon.

**Comment by pgawlowicz** [comment]
@apperezG Please check our updated [tutorial](https://docs.srsran.com/projects/project/en/latest/tutorials/source/flexric/source/index.html), where we use the `br-flexric` branch that allows specifying metric names using a config file that is passed to a xApp.

**Reply by apperezG** [reply]
Good morning @pgawlowicz ! 

Thank you for the new tutorial, it's much easier!
However, I encountered some problems working with the example. The output should be like this (requesting the DRB.UEThpDl and DRB.UEThpUl metrics):
![Screenshot 2024-01-26 at 12-39-39 O-RAN NearRT-RIC and xApp — srsRAN Project documentation](https://github.com/srsran/srsRAN_Project/assets/107911736/cf88cb89-e608-42e3-9476-ad70b0f28b05)
My configuration file is this one: 
```
SM_DIR = "/usr/local/lib/flexric/"

# supported name = xApp
Name = "xApp"
NearRT_RIC_IP = "192.168.20.2"
E42_Port = 36422

# supported name : KPM
# supported time (ms): 1000
# supported format: 1
# supported ran_type: ngran_gNB
Sub_ORAN_SM_List = (
    { name = "KPM", time = 1000,
      format = 1,
      ran_type = "ngran_gNB",
      actions = (
            { name = "CQI" },
            { name = "RSRP" },
            { name = "DRB.UEThpDl" },
            { name = "DRB.UEThpUl" }
            )
    }
)

xApp_DB = {
    enable = "OFF"
    ip = "127.0.0.1"
    dir = "/tmp/"
    filename = "testdb"
    username = "your_username" # if using mysql
    password = "your_passwd" # if using mysql
}
```

And my output is this one: 
![logs](https://github.com/srsran/srsRAN_Project/assets/107911736/dd65ec4d-7132-4a06-bd86-af264f508f46)

Here you can see how the CQI and RSRP values are returned correctly but the DRB.UEThpDl and DRB.UEThpU values are not.

**Comment by pgawlowicz** [comment]
What does happen if you request only the two metrics in the xApp config file?
```
            { name = "DRB.UEThpDl" },
            { name = "DRB.UEThpUl" }
```

Also, did you try sending more UL data, for example with the following command:
```
sudo ip netns exec ue1 iperf -c 10.45.1.1 -u -b 10M -i 1 -t 60
```

**Reply by apperezG** [reply]
Testing what you are telling me gives me the following result:
![flexric_dl_ul](https://github.com/srsran/srsRAN_Project/assets/107911736/580e58e1-d9b6-4bb8-a594-0a43a1727f36)
The value of DRB.UEThpDl is always 0 and the value of DRB.UEThpUl is always 11943.

**Reply by pgawlowicz** [reply]
Is there any change if you disable iperf?

BTW. you are getting a lot of RF issues, did you run [srsran_performance](https://github.com/srsran/srsRAN_Project/blob/main/scripts/srsran_performance) on both gnb and srsUE hosts?

**Reply by apperezG** [reply]
Executing a simple ping I have the following output:
![monitor_ping](https://github.com/srsran/srsRAN_Project/assets/107911736/67d6c286-5e3a-4eb8-9257-4ac3e57aaccd)

I executed the srsran_performance and the xApp gives me the same output.

**Reply by pgawlowicz** [reply]
could you send ping more frequently with `-i 0.1` option? I think the traffic volume is simply too low. And the metric unit is kbps.

**Reply by apperezG** [reply]
The metric is still kept at 0 despite the addition of the option

**Reply by pgawlowicz** [reply]
could you tell me the command you use to ping? also could you print the routing table on this host? using `route -n`
